src/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -59,27 +59,60 @@ In turn these random seeds are used by:
 
 This randomness may be drawn from various Filecoin chain epochs by the respective protocols that use them according to their security requirements.
 
-It is important to note that a given Filecoin network and a given drand network need not have the same round time, i.e. blocks may be generated faster or slower by Filecoin than randomness is generated by drand.
-For instance, if the drand beacon is producing randomness twice as fast as Filecoin produces blocks, we might expect two random values to be produced in a Filecoin epoch, conversely if the Filecoin network is twice as fast as drand, we might expect a random value every other Filecoin epoch. Accordingly, depending on both networks' configurations, certain Filecoin blocks could contain multiple or no drand entries.
+It is important to note that a given Filecoin network and a given drand network
+need not have the same round time, i.e. blocks may be generated faster or slower
+by Filecoin than randomness is generated by drand.  For instance, if the drand
+beacon is producing randomness twice as fast as Filecoin produces blocks, we
+might expect two random values to be produced in a Filecoin epoch, conversely if
+the Filecoin network is twice as fast as drand, we might expect a random value
+every other Filecoin epoch. Accordingly, depending on both networks'
+configurations, certain Filecoin blocks could contain multiple or no drand
+entries.
+
+In all cases, Filecoin blocks must include all drand beacon outputs generated
+since the last epoch in the `BeaconEntries` field of the block header. Any use
+of randomness from a given Filecoin epoch should use the last valid drand entry
+included in a Filecoin block, as follows:
+
+### Get drand randomness for VM
+
+For operations such as Porep creation, proofs validations, anything that
+requires randomness from the VM point of view, the following method shows how to
+extract the drand entrie from the chain itself.
+```go
+func GetRandomnessSeed(e ChainEpoch) {
+  // get the minimum epoch number in this period associated with the requested epoch
+  // so if election period is 2, e = 11, then this function returns 10
+  minEpoch := getFirstEpochFromPeriod(e)
+  // get the drand round associated with this election period
+  // defined in the drand section
+  drandRound := MaxBeaconRoundForEpoch(minEpoch)
+  for {
+     // if this is not a null block, then it must have the entry we want
+    if !chain.IsNullBlock(minEpoch) 
```

Comment:
I don't understand what you mean - I just put this `IsNullBlock` because I don't know where / if it's implemented somewhere. Do you mean that I should write down the spec of `IsNullBlock` ?

Reply:
no it was just a design opinion with regards to where this call should be made, but can't trace what I meant anymore...